riscv/triggers.cc

```diff
@@ -190,7 +190,8 @@ std::optional<match_result_t> mcontrol_common_t::detect_memory_access_match(proc
   if ((operation == triggers::OPERATION_EXECUTE && !execute) ||
       (operation == triggers::OPERATION_STORE && !store) ||
       (operation == triggers::OPERATION_LOAD && !load) ||
-      !mode_match(proc->get_state())) {
+      !mode_match(proc->get_state()) ||
+      !textra_match(proc)) {
```

Comment:
Why not combine these two methods? They are always called together.

Reply:
I prefer to separate the ```mode_match()``` and ```textra_match()``` for the following reasons.
1. The ```(!mode_match(proc->get_state()) || !textra_match(proc))``` clarifies two independent checkings and, thus, has a higher readability than ```(!mode_and_textra_match(proc))```.
2. The current separated version of ```mode_match()``` is more elegant than ```mode_and_textra_match()```.
> bool trigger_t::mode_match(state_t * const state) const noexcept
> {
>   switch (state->prv) {
>     case PRV_M: return m;
>     case PRV_S: return state->v ? vs : s;
>     case PRV_U: return state->v ? vu : u;
>     default: assert(false);
>   }
> }

The ```mode_and_textra_match()``` will go with something like:

>   switch (state->prv) {
>     case PRV_M: if (!m) return false;
>     case PRV_S: if (!(state->v ? vs : s)) return false;
>     case PRV_U: if (!(state->v ? vu : u)) return false;
>     default: assert(false);
>   }
>   ...

3. The ```textra_match()``` is already a structurally complicated function. I want to keep it conceptually simple, i.e., checking textra CSR.

With the above concerns, how about adding a new function ```trigger_t::common_match()```?
```
bool trigger_t::common_match(processor_t * const proc) const noexcept 
{ return mode_match(proc->get_state()) && textra_match(proc); }
```

Reply:
I'm impressed with your reasoning. Let's leave it like it is.